In [159]:
import numpy as np
from datascience import *


import matplotlib
%matplotlib inline
import matplotlib.pyplot as plt
plt.style.use('fivethirtyeight')
import warnings
warnings.simplefilter('ignore', FutureWarning)
from scipy.stats import zscore

boxScore = Table.read_table('2017_2018_season.csv', sep=',')
boxScore.show(10)

In [160]:
contracts = Table.read_table('2018_2019_Contracts.csv', sep=',')
contracts.show(10)

In [161]:
boxscore_clean = Table.read_table('2017_2018_season(clean).csv', sep=',')
boxscore_clean.show(10)

In [162]:
rookies_17 = Table.read_table('rookies_2017-18.csv', sep=',')
rookies_17.show(10)

In [163]:
joined_17 = boxscore_clean.join('Player', rookies_17, 'Unnamed: 1')

In [164]:
rookies_names_17 = joined_17.column(0)

In [165]:
rookies_names_17.item(len(rookies_names_17)-1)

'Zhou Qi\\qizh01'

In [166]:
for name in np.arange(len(rookies_names_17)-1):
    boxscore_clean = boxscore_clean.where('Player', are.not_equal_to(rookies_names_17.item(name)))

In [167]:
boxscore_clean

Rk   | Player                      | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G
1    | Alex Abrines\abrinal01      | SG   | 24   | OKC  | 75   | 8    | 15.1 | 1.5  | 3.9  | 0.395 | 1.1  | 2.9  | 0.38  | 0.4  | 0.9  | 0.443 | 0.54  | 0.5  | 0.6  | 0.848 | 0.3  | 1.2  | 1.5  | 0.4  | 0.5  | 0.1  | 0.3  | 1.7  | 4.7
2    | Quincy Acy\acyqu01          | PF   | 27   | BRK  | 70   | 8    | 19.4 | 1.9  | 5.2  | 0.356 | 1.5  | 4.2  | 0.349 | 0.4  | 1    | 0.384 | 0.496 | 0.7  | 0.9  | 0.817 | 0.6  | 3.1  | 3.7  | 0.8  | 0.5  | 0.4  | 0.9  | 2.1  | 5.9
3    | Steven Adams\adamsst01      | C    | 24   | OKC  | 76   | 76   | 32.7 | 5.9  | 9.4  | 0.629 | 0    | 0    | 0     | 5.9  | 9.3  | 0.631 | 0.629 | 2.1  | 3.8  | 0.559 | 5.1  | 4    | 9    | 1.2  | 1.2  | 1    | 1.7  | 2.8  | 13.9
5    | Arron Afflalo\afflaar01     | SG   | 32   | ORL  | 53   | 3    | 12.9 | 1.2  | 3.1  | 0.401 | 0.5  | 1.3  | 0.386 | 0.7  | 1.7  | 0.413 | 0.485 | 0.4  | 0.5  | 0.846 | 0.1  | 1.2  | 1.2  | 0.6  | 0.1  | 0.2  | 0.4  | 1.1  | 3.4
6    | Cole Aldrich\aldrico01      | C    | 29   | MIN  | 21   | 0    | 2.3  | 0.2  | 0.7  | 0.333 | 0    | 0    | nan   | 0.2  | 0.7  | 0.333 | 0.333 | 0.1  | 0.3  | 0.333 | 0.1  | 0.6  | 0.7  | 0.1  | 0.1  | 0    | 0    | 0.5  | 0.6
7    | LaMarcus Aldridge\aldrila01 | C    | 32   | SAS  | 75   | 75   | 33.5 | 9.2  | 18   | 0.51  | 0.4  | 1.2  | 0.293 | 8.8  | 16.7 | 0.526 | 0.52  | 4.5  | 5.3  | 0.837 | 3.3  | 5.2  | 8.5  | 2    | 0.6  | 1.2  | 1.5  | 2.1  | 23.1
10   | Tony Allen\allento01        | SF   | 36   | NOP  | 22   | 0    | 12.4 | 2    | 4.1  | 0.484 | 0.2  | 0.5  | 0.333 | 1.8  | 3.6  | 0.506 | 0.505 | 0.5  | 1    | 0.524 | 0.9  | 1.2  | 2.1  | 0.4  | 0.5  | 0.1  | 0.9  | 2.2  | 4.7
11   | Al-Farouq Aminu\aminual01   | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3
12   | Justin Anderson\anderju01   | SF   | 24   | PHI  | 38   | 0    | 13.7 | 2.3  | 5.3  | 0.431 | 0.9  | 2.7  | 0.33  | 1.4  | 2.6  | 0.535 | 0.515 | 0.7  | 1    | 0.737 | 0.7  | 1.8  | 2.4  | 0.7  | 0.4  | 0.2  | 0.4  | 1.4  | 6.2
13   | Kyle Anderson\anderky01     | SF   | 24   | SAS  | 74   | 67   | 26.7 | 3.1  | 5.9  | 0.527 | 0.3  | 0.8  | 0.333 | 2.9  | 5.1  | 0.556 | 0.549 | 1.4  | 2    | 0.712 | 1.1  | 4.2  | 5.4  | 2.7  | 1.6  | 0.8  | 1.3  | 1.5  | 7.9
... (538 rows omitted)

In [168]:
rookies_16 = Table.read_table('rookies_2016-17.csv', sep=',')
joined_16 = boxscore_clean.join('Player', rookies_16, 'Player')
rookies_names_16 = joined_16.column(0)
for name in np.arange(len(rookies_names_16)-1):
    boxscore_clean = boxscore_clean.where('Player', are.not_equal_to(rookies_names_16.item(name)))
boxscore_clean

Rk   | Player                      | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G
2    | Quincy Acy\acyqu01          | PF   | 27   | BRK  | 70   | 8    | 19.4 | 1.9  | 5.2  | 0.356 | 1.5  | 4.2  | 0.349 | 0.4  | 1    | 0.384 | 0.496 | 0.7  | 0.9  | 0.817 | 0.6  | 3.1  | 3.7  | 0.8  | 0.5  | 0.4  | 0.9  | 2.1  | 5.9
3    | Steven Adams\adamsst01      | C    | 24   | OKC  | 76   | 76   | 32.7 | 5.9  | 9.4  | 0.629 | 0    | 0    | 0     | 5.9  | 9.3  | 0.631 | 0.629 | 2.1  | 3.8  | 0.559 | 5.1  | 4    | 9    | 1.2  | 1.2  | 1    | 1.7  | 2.8  | 13.9
5    | Arron Afflalo\afflaar01     | SG   | 32   | ORL  | 53   | 3    | 12.9 | 1.2  | 3.1  | 0.401 | 0.5  | 1.3  | 0.386 | 0.7  | 1.7  | 0.413 | 0.485 | 0.4  | 0.5  | 0.846 | 0.1  | 1.2  | 1.2  | 0.6  | 0.1  | 0.2  | 0.4  | 1.1  | 3.4
6    | Cole Aldrich\aldrico01      | C    | 29   | MIN  | 21   | 0    | 2.3  | 0.2  | 0.7  | 0.333 | 0    | 0    | nan   | 0.2  | 0.7  | 0.333 | 0.333 | 0.1  | 0.3  | 0.333 | 0.1  | 0.6  | 0.7  | 0.1  | 0.1  | 0    | 0    | 0.5  | 0.6
7    | LaMarcus Aldridge\aldrila01 | C    | 32   | SAS  | 75   | 75   | 33.5 | 9.2  | 18   | 0.51  | 0.4  | 1.2  | 0.293 | 8.8  | 16.7 | 0.526 | 0.52  | 4.5  | 5.3  | 0.837 | 3.3  | 5.2  | 8.5  | 2    | 0.6  | 1.2  | 1.5  | 2.1  | 23.1
10   | Tony Allen\allento01        | SF   | 36   | NOP  | 22   | 0    | 12.4 | 2    | 4.1  | 0.484 | 0.2  | 0.5  | 0.333 | 1.8  | 3.6  | 0.506 | 0.505 | 0.5  | 1    | 0.524 | 0.9  | 1.2  | 2.1  | 0.4  | 0.5  | 0.1  | 0.9  | 2.2  | 4.7
11   | Al-Farouq Aminu\aminual01   | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3
12   | Justin Anderson\anderju01   | SF   | 24   | PHI  | 38   | 0    | 13.7 | 2.3  | 5.3  | 0.431 | 0.9  | 2.7  | 0.33  | 1.4  | 2.6  | 0.535 | 0.515 | 0.7  | 1    | 0.737 | 0.7  | 1.8  | 2.4  | 0.7  | 0.4  | 0.2  | 0.4  | 1.4  | 6.2
13   | Kyle Anderson\anderky01     | SF   | 24   | SAS  | 74   | 67   | 26.7 | 3.1  | 5.9  | 0.527 | 0.3  | 0.8  | 0.333 | 2.9  | 5.1  | 0.556 | 0.549 | 1.4  | 2    | 0.712 | 1.1  | 4.2  | 5.4  | 2.7  | 1.6  | 0.8  | 1.3  | 1.5  | 7.9
14   | Ryan Anderson\anderry01     | PF   | 29   | HOU  | 66   | 50   | 26.1 | 3.1  | 7.3  | 0.431 | 2    | 5.1  | 0.386 | 1.2  | 2.1  | 0.539 | 0.568 | 1.1  | 1.4  | 0.774 | 1.4  | 3.6  | 5    | 0.9  | 0.4  | 0.3  | 0.6  | 1.9  | 9.3
... (468 rows omitted)

In [169]:
rookies_15 = Table.read_table('rookies_2015-16.csv', sep=',')
joined_15 = boxscore_clean.join('Player', rookies_15, 'Player')
rookies_names_15 = joined_15.column(0)
for name in np.arange(len(rookies_names_15)-1):
    boxscore_clean = boxscore_clean.where('Player', are.not_equal_to(rookies_names_15.item(name)))
boxscore_clean

Rk   | Player                          | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G
2    | Quincy Acy\acyqu01              | PF   | 27   | BRK  | 70   | 8    | 19.4 | 1.9  | 5.2  | 0.356 | 1.5  | 4.2  | 0.349 | 0.4  | 1    | 0.384 | 0.496 | 0.7  | 0.9  | 0.817 | 0.6  | 3.1  | 3.7  | 0.8  | 0.5  | 0.4  | 0.9  | 2.1  | 5.9
3    | Steven Adams\adamsst01          | C    | 24   | OKC  | 76   | 76   | 32.7 | 5.9  | 9.4  | 0.629 | 0    | 0    | 0     | 5.9  | 9.3  | 0.631 | 0.629 | 2.1  | 3.8  | 0.559 | 5.1  | 4    | 9    | 1.2  | 1.2  | 1    | 1.7  | 2.8  | 13.9
5    | Arron Afflalo\afflaar01         | SG   | 32   | ORL  | 53   | 3    | 12.9 | 1.2  | 3.1  | 0.401 | 0.5  | 1.3  | 0.386 | 0.7  | 1.7  | 0.413 | 0.485 | 0.4  | 0.5  | 0.846 | 0.1  | 1.2  | 1.2  | 0.6  | 0.1  | 0.2  | 0.4  | 1.1  | 3.4
6    | Cole Aldrich\aldrico01          | C    | 29   | MIN  | 21   | 0    | 2.3  | 0.2  | 0.7  | 0.333 | 0    | 0    | nan   | 0.2  | 0.7  | 0.333 | 0.333 | 0.1  | 0.3  | 0.333 | 0.1  | 0.6  | 0.7  | 0.1  | 0.1  | 0    | 0    | 0.5  | 0.6
7    | LaMarcus Aldridge\aldrila01     | C    | 32   | SAS  | 75   | 75   | 33.5 | 9.2  | 18   | 0.51  | 0.4  | 1.2  | 0.293 | 8.8  | 16.7 | 0.526 | 0.52  | 4.5  | 5.3  | 0.837 | 3.3  | 5.2  | 8.5  | 2    | 0.6  | 1.2  | 1.5  | 2.1  | 23.1
10   | Tony Allen\allento01            | SF   | 36   | NOP  | 22   | 0    | 12.4 | 2    | 4.1  | 0.484 | 0.2  | 0.5  | 0.333 | 1.8  | 3.6  | 0.506 | 0.505 | 0.5  | 1    | 0.524 | 0.9  | 1.2  | 2.1  | 0.4  | 0.5  | 0.1  | 0.9  | 2.2  | 4.7
11   | Al-Farouq Aminu\aminual01       | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3
13   | Kyle Anderson\anderky01         | SF   | 24   | SAS  | 74   | 67   | 26.7 | 3.1  | 5.9  | 0.527 | 0.3  | 0.8  | 0.333 | 2.9  | 5.1  | 0.556 | 0.549 | 1.4  | 2    | 0.712 | 1.1  | 4.2  | 5.4  | 2.7  | 1.6  | 0.8  | 1.3  | 1.5  | 7.9
14   | Ryan Anderson\anderry01         | PF   | 29   | HOU  | 66   | 50   | 26.1 | 3.1  | 7.3  | 0.431 | 2    | 5.1  | 0.386 | 1.2  | 2.1  | 0.539 | 0.568 | 1.1  | 1.4  | 0.774 | 1.4  | 3.6  | 5    | 0.9  | 0.4  | 0.3  | 0.6  | 1.9  | 9.3
16   | Giannis Antetokounmpo\antetgi01 | PF   | 23   | MIL  | 75   | 75   | 36.7 | 9.9  | 18.7 | 0.529 | 0.6  | 1.9  | 0.307 | 9.3  | 16.8 | 0.554 | 0.545 | 6.5  | 8.5  | 0.76  | 2.1  | 8    | 10   | 4.8  | 1.5  | 1.4  | 3    | 3.1  | 26.9
... (415 rows omitted)

In [170]:
boxscore_clean

Rk   | Player                          | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G
2    | Quincy Acy\acyqu01              | PF   | 27   | BRK  | 70   | 8    | 19.4 | 1.9  | 5.2  | 0.356 | 1.5  | 4.2  | 0.349 | 0.4  | 1    | 0.384 | 0.496 | 0.7  | 0.9  | 0.817 | 0.6  | 3.1  | 3.7  | 0.8  | 0.5  | 0.4  | 0.9  | 2.1  | 5.9
3    | Steven Adams\adamsst01          | C    | 24   | OKC  | 76   | 76   | 32.7 | 5.9  | 9.4  | 0.629 | 0    | 0    | 0     | 5.9  | 9.3  | 0.631 | 0.629 | 2.1  | 3.8  | 0.559 | 5.1  | 4    | 9    | 1.2  | 1.2  | 1    | 1.7  | 2.8  | 13.9
5    | Arron Afflalo\afflaar01         | SG   | 32   | ORL  | 53   | 3    | 12.9 | 1.2  | 3.1  | 0.401 | 0.5  | 1.3  | 0.386 | 0.7  | 1.7  | 0.413 | 0.485 | 0.4  | 0.5  | 0.846 | 0.1  | 1.2  | 1.2  | 0.6  | 0.1  | 0.2  | 0.4  | 1.1  | 3.4
6    | Cole Aldrich\aldrico01          | C    | 29   | MIN  | 21   | 0    | 2.3  | 0.2  | 0.7  | 0.333 | 0    | 0    | nan   | 0.2  | 0.7  | 0.333 | 0.333 | 0.1  | 0.3  | 0.333 | 0.1  | 0.6  | 0.7  | 0.1  | 0.1  | 0    | 0    | 0.5  | 0.6
7    | LaMarcus Aldridge\aldrila01     | C    | 32   | SAS  | 75   | 75   | 33.5 | 9.2  | 18   | 0.51  | 0.4  | 1.2  | 0.293 | 8.8  | 16.7 | 0.526 | 0.52  | 4.5  | 5.3  | 0.837 | 3.3  | 5.2  | 8.5  | 2    | 0.6  | 1.2  | 1.5  | 2.1  | 23.1
10   | Tony Allen\allento01            | SF   | 36   | NOP  | 22   | 0    | 12.4 | 2    | 4.1  | 0.484 | 0.2  | 0.5  | 0.333 | 1.8  | 3.6  | 0.506 | 0.505 | 0.5  | 1    | 0.524 | 0.9  | 1.2  | 2.1  | 0.4  | 0.5  | 0.1  | 0.9  | 2.2  | 4.7
11   | Al-Farouq Aminu\aminual01       | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3
13   | Kyle Anderson\anderky01         | SF   | 24   | SAS  | 74   | 67   | 26.7 | 3.1  | 5.9  | 0.527 | 0.3  | 0.8  | 0.333 | 2.9  | 5.1  | 0.556 | 0.549 | 1.4  | 2    | 0.712 | 1.1  | 4.2  | 5.4  | 2.7  | 1.6  | 0.8  | 1.3  | 1.5  | 7.9
14   | Ryan Anderson\anderry01         | PF   | 29   | HOU  | 66   | 50   | 26.1 | 3.1  | 7.3  | 0.431 | 2    | 5.1  | 0.386 | 1.2  | 2.1  | 0.539 | 0.568 | 1.1  | 1.4  | 0.774 | 1.4  | 3.6  | 5    | 0.9  | 0.4  | 0.3  | 0.6  | 1.9  | 9.3
16   | Giannis Antetokounmpo\antetgi01 | PF   | 23   | MIL  | 75   | 75   | 36.7 | 9.9  | 18.7 | 0.529 | 0.6  | 1.9  | 0.307 | 9.3  | 16.8 | 0.554 | 0.545 | 6.5  | 8.5  | 0.76  | 2.1  | 8    | 10   | 4.8  | 1.5  | 1.4  | 3    | 3.1  | 26.9
... (415 rows omitted)

In [171]:
contracts

Rk   | Player                      | Tm   | 2018-19   | 2019-20   | 2020-21   | 2021-22   | 2022-23   | 2023-24 | Signed Using   | Guaranteed
1    | Stephen Curry\curryst01     | GSW  | $37457154 | $40231758 | $43006362 | $45780966 | nan       | nan     | Bird Rights    | $166476240
2    | Chris Paul\paulch01         | HOU  | $35654150 | $38506482 | $41358814 | $44211146 | nan       | nan     | nan            | $159730592
3    | Russell Westbrook\westbru01 | OKC  | $35654150 | $38178000 | $41006000 | $43848000 | $46662000 | nan     | Bird Rights    | $158686150
4    | LeBron James\jamesle01      | LAL  | $35654150 | $37436858 | $39219565 | $41002273 | nan       | nan     | nan            | $113310573
5    | Blake Griffin\griffbl01     | DET  | $32088932 | $34234964 | $36595996 | $38957028 | nan       | nan     | Bird Rights    | $102919892
6    | Gordon Hayward\haywago01    | BOS  | $31214295 | $32700690 | $34187085 | nan       | nan       | nan     | Cap space      | $63914985
7    | Kyle Lowry\lowryky01        | TOR  | $31200000 | $33296296 | nan       | nan       | nan       | nan     | Bird Rights    | $64496296
8    | Paul George\georgpa01       | OKC  | $30560700 | $33005556 | $35450412 | $37895268 | nan       | nan     | Maximum Salary | $99016668
9    | Mike Conley\conlemi01       | MEM  | $30521116 | $32511623 | $34504132 | nan       | nan       | nan     | Cap Space      | $85459122
10   | James Harden\hardeja01      | HOU  | $30431854 | $37800000 | $40824000 | $43848000 | $46872000 | nan     | Bird Rights    | $152903854
... (478 rows omitted)

In [172]:
full_clean_table = boxscore_clean.join('Player', contracts, 'Player')

In [173]:
full_clean_table

Player                    | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19   | 2019-20   | 2020-21   | 2021-22   | 2022-23 | 2023-24 | Signed Using   | Guaranteed
Aaron Gordon\gordoaa01    | 179  | PF   | 22   | ORL  | 58   | 57   | 32.9 | 6.5  | 14.9 | 0.434 | 2    | 5.9  | 0.336 | 4.5  | 9    | 0.497 | 0.5   | 2.7  | 3.9  | 0.698 | 1.5  | 6.4  | 7.9  | 2.3  | 1    | 0.8  | 1.8  | 1.9  | 17.6 | 40   | ORL  | $21590909 | $19863636 | $18136364 | $16409091 | nan     | nan     | nan            | $76000000
Al Horford\horfoal01      | 227  | C    | 31   | BOS  | 72   | 72   | 31.6 | 5.1  | 10.5 | 0.489 | 1.3  | 3.1  | 0.429 | 3.8  | 7.3  | 0.514 | 0.553 | 1.3  | 1.7  | 0.783 | 1.4  | 5.9  | 7.4  | 4.7  | 0.6  | 1.1  | 1.8  | 1.9  | 12.9 | 13   | BOS  | $28928710 | $30123015 | nan       | nan       | nan     | nan     | Cap Space      | $28928710
Al Jefferson\jeffeal01    | 250  | C    | 33   | IND  | 36   | 1    | 13.4 | 3.1  | 5.8  | 0.534 | 0    | 0.1  | 0     | 3.1  | 5.7  | 0.541 | 0.534 | 0.8  | 1    | 0.833 | 0.8  | 3.1  | 4    | 0.8  | 0.4  | 0.6  | 0.6  | 1.8  | 7    | 235  | IND  | $4000000  | nan       | nan       | nan       | nan     | nan     | nan            | $4000000
Al-Farouq Aminu\aminual01 | 11   | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3  | 179  | POR  | $6957105  | nan       | nan       | nan       | nan     | nan     | Cap Space      | $6957105
Alec Burks\burksal01      | 78   | SG   | 26   | UTA  | 64   | 1    | 16.5 | 2.7  | 6.5  | 0.411 | 0.7  | 2.2  | 0.331 | 1.9  | 4.3  | 0.452 | 0.467 | 1.7  | 1.9  | 0.863 | 0.3  | 2.7  | 3    | 1    | 0.6  | 0.1  | 0.9  | 1.2  | 7.7  | 120  | UTA  | $11536515 | nan       | nan       | nan       | nan     | nan     | 1st Round Pick | $11536515
Alex Len\lenal01          | 292  | C    | 24   | PHO  | 69   | 13   | 20.2 | 3.1  | 5.5  | 0.566 | 0    | 0    | 0.333 | 3.1  | 5.4  | 0.568 | 0.567 | 2.3  | 3.3  | 0.684 | 2.5  | 5    | 7.5  | 1.2  | 0.4  | 0.9  | 1.1  | 2.3  | 8.5  | 227  | ATL  | $4350000  | $4160000  | nan       | nan       | nan     | nan     | Room Exception | $8510000
Allen Crabbe\crabbal01    | 114  | SG   | 25   | BRK  | 75   | 68   | 29.3 | 4.5  | 11   | 0.407 | 2.7  | 7.1  | 0.378 | 1.8  | 3.9  | 0.461 | 0.529 | 1.5  | 1.8  | 0.852 | 0.4  | 3.9  | 4.3  | 1.6  | 0.6  | 0.5  | 1    | 2.2  | 13.2 | 57   | BRK  | $18500000 | $18500000 | nan       | nan       | nan     | nan     | Cap Space      | $18500000
Amir Johnson\johnsam01    | 254  | C    | 30   | PHI  | 74   | 18   | 15.8 | 1.9  | 3.5  | 0.538 | 0.1  | 0.4  | 0.313 | 1.8  | 3.1  | 0.57  | 0.558 | 0.7  | 1.1  | 0.612 | 1.7  | 2.8  | 4.5  | 1.6  | 0.6  | 0.6  | 0.7  | 2.6  | 4.6  | 301  | PHI  | $2393887  | nan       | nan       | nan       | nan     | nan     | Minimum Salary | $2393887
Andre Drummond\drumman01  | 140  | C    | 24   | DET  | 78   | 78   | 33.7 | 6    | 11.3 | 0.529 | 0    | 0.1  | 0     | 6    | 11.2 | 0.536 | 0.529 | 3.1  | 5.1  | 0.605 | 5.1  | 10.9 | 16   | 3    | 1.5  | 1.6  | 2.6  | 3.2  | 15   | 25   | DET  | $25434263 | $27093019 | $28751775 | nan       | nan     | nan     | Cap Space      | $52527282
Andre Iguodala\iguodan01  | 234  | SF   | 34   | GSW  | 64   | 7    | 25.3 | 2.3  | 5    | 0.463 | 0.5  | 1.8  | 0.282 | 1.8  | 3.2  | 0.567 | 0.514 | 0.9  | 1.4  | 0.632 | 0.8  | 3.1  | 3.8  | 3.3  | 0.8  | 0.6  | 1    | 1.5  | 6    | 71   | GSW  | $16000000 | $17185185 | nan       | nan       | nan     | nan     | Bird Rights    | $33185185
... (361 rows omitted)

In [174]:
clean_data = full_clean_table.drop('2019-20').drop('2020-21').drop('2021-22').drop('2022-23').drop('2023-24').drop('Signed Using').drop('Guaranteed')

In [175]:
import pandas as pd

In [176]:
clean_data

Player                    | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Aaron Gordon\gordoaa01    | 179  | PF   | 22   | ORL  | 58   | 57   | 32.9 | 6.5  | 14.9 | 0.434 | 2    | 5.9  | 0.336 | 4.5  | 9    | 0.497 | 0.5   | 2.7  | 3.9  | 0.698 | 1.5  | 6.4  | 7.9  | 2.3  | 1    | 0.8  | 1.8  | 1.9  | 17.6 | 40   | ORL  | $21590909
Al Horford\horfoal01      | 227  | C    | 31   | BOS  | 72   | 72   | 31.6 | 5.1  | 10.5 | 0.489 | 1.3  | 3.1  | 0.429 | 3.8  | 7.3  | 0.514 | 0.553 | 1.3  | 1.7  | 0.783 | 1.4  | 5.9  | 7.4  | 4.7  | 0.6  | 1.1  | 1.8  | 1.9  | 12.9 | 13   | BOS  | $28928710
Al Jefferson\jeffeal01    | 250  | C    | 33   | IND  | 36   | 1    | 13.4 | 3.1  | 5.8  | 0.534 | 0    | 0.1  | 0     | 3.1  | 5.7  | 0.541 | 0.534 | 0.8  | 1    | 0.833 | 0.8  | 3.1  | 4    | 0.8  | 0.4  | 0.6  | 0.6  | 1.8  | 7    | 235  | IND  | $4000000
Al-Farouq Aminu\aminual01 | 11   | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3  | 179  | POR  | $6957105
Alec Burks\burksal01      | 78   | SG   | 26   | UTA  | 64   | 1    | 16.5 | 2.7  | 6.5  | 0.411 | 0.7  | 2.2  | 0.331 | 1.9  | 4.3  | 0.452 | 0.467 | 1.7  | 1.9  | 0.863 | 0.3  | 2.7  | 3    | 1    | 0.6  | 0.1  | 0.9  | 1.2  | 7.7  | 120  | UTA  | $11536515
Alex Len\lenal01          | 292  | C    | 24   | PHO  | 69   | 13   | 20.2 | 3.1  | 5.5  | 0.566 | 0    | 0    | 0.333 | 3.1  | 5.4  | 0.568 | 0.567 | 2.3  | 3.3  | 0.684 | 2.5  | 5    | 7.5  | 1.2  | 0.4  | 0.9  | 1.1  | 2.3  | 8.5  | 227  | ATL  | $4350000
Allen Crabbe\crabbal01    | 114  | SG   | 25   | BRK  | 75   | 68   | 29.3 | 4.5  | 11   | 0.407 | 2.7  | 7.1  | 0.378 | 1.8  | 3.9  | 0.461 | 0.529 | 1.5  | 1.8  | 0.852 | 0.4  | 3.9  | 4.3  | 1.6  | 0.6  | 0.5  | 1    | 2.2  | 13.2 | 57   | BRK  | $18500000
Amir Johnson\johnsam01    | 254  | C    | 30   | PHI  | 74   | 18   | 15.8 | 1.9  | 3.5  | 0.538 | 0.1  | 0.4  | 0.313 | 1.8  | 3.1  | 0.57  | 0.558 | 0.7  | 1.1  | 0.612 | 1.7  | 2.8  | 4.5  | 1.6  | 0.6  | 0.6  | 0.7  | 2.6  | 4.6  | 301  | PHI  | $2393887
Andre Drummond\drumman01  | 140  | C    | 24   | DET  | 78   | 78   | 33.7 | 6    | 11.3 | 0.529 | 0    | 0.1  | 0     | 6    | 11.2 | 0.536 | 0.529 | 3.1  | 5.1  | 0.605 | 5.1  | 10.9 | 16   | 3    | 1.5  | 1.6  | 2.6  | 3.2  | 15   | 25   | DET  | $25434263
Andre Iguodala\iguodan01  | 234  | SF   | 34   | GSW  | 64   | 7    | 25.3 | 2.3  | 5    | 0.463 | 0.5  | 1.8  | 0.282 | 1.8  | 3.2  | 0.567 | 0.514 | 0.9  | 1.4  | 0.632 | 0.8  | 3.1  | 3.8  | 3.3  | 0.8  | 0.6  | 1    | 1.5  | 6    | 71   | GSW  | $16000000
... (361 rows omitted)

In [177]:
def convert_string_to_number(string):
    return float(string.strip('$'))*1

In [178]:
convert_string_to_number('$21590909')

21590909.0

In [179]:
clean_data

Player                    | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Aaron Gordon\gordoaa01    | 179  | PF   | 22   | ORL  | 58   | 57   | 32.9 | 6.5  | 14.9 | 0.434 | 2    | 5.9  | 0.336 | 4.5  | 9    | 0.497 | 0.5   | 2.7  | 3.9  | 0.698 | 1.5  | 6.4  | 7.9  | 2.3  | 1    | 0.8  | 1.8  | 1.9  | 17.6 | 40   | ORL  | $21590909
Al Horford\horfoal01      | 227  | C    | 31   | BOS  | 72   | 72   | 31.6 | 5.1  | 10.5 | 0.489 | 1.3  | 3.1  | 0.429 | 3.8  | 7.3  | 0.514 | 0.553 | 1.3  | 1.7  | 0.783 | 1.4  | 5.9  | 7.4  | 4.7  | 0.6  | 1.1  | 1.8  | 1.9  | 12.9 | 13   | BOS  | $28928710
Al Jefferson\jeffeal01    | 250  | C    | 33   | IND  | 36   | 1    | 13.4 | 3.1  | 5.8  | 0.534 | 0    | 0.1  | 0     | 3.1  | 5.7  | 0.541 | 0.534 | 0.8  | 1    | 0.833 | 0.8  | 3.1  | 4    | 0.8  | 0.4  | 0.6  | 0.6  | 1.8  | 7    | 235  | IND  | $4000000
Al-Farouq Aminu\aminual01 | 11   | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3  | 179  | POR  | $6957105
Alec Burks\burksal01      | 78   | SG   | 26   | UTA  | 64   | 1    | 16.5 | 2.7  | 6.5  | 0.411 | 0.7  | 2.2  | 0.331 | 1.9  | 4.3  | 0.452 | 0.467 | 1.7  | 1.9  | 0.863 | 0.3  | 2.7  | 3    | 1    | 0.6  | 0.1  | 0.9  | 1.2  | 7.7  | 120  | UTA  | $11536515
Alex Len\lenal01          | 292  | C    | 24   | PHO  | 69   | 13   | 20.2 | 3.1  | 5.5  | 0.566 | 0    | 0    | 0.333 | 3.1  | 5.4  | 0.568 | 0.567 | 2.3  | 3.3  | 0.684 | 2.5  | 5    | 7.5  | 1.2  | 0.4  | 0.9  | 1.1  | 2.3  | 8.5  | 227  | ATL  | $4350000
Allen Crabbe\crabbal01    | 114  | SG   | 25   | BRK  | 75   | 68   | 29.3 | 4.5  | 11   | 0.407 | 2.7  | 7.1  | 0.378 | 1.8  | 3.9  | 0.461 | 0.529 | 1.5  | 1.8  | 0.852 | 0.4  | 3.9  | 4.3  | 1.6  | 0.6  | 0.5  | 1    | 2.2  | 13.2 | 57   | BRK  | $18500000
Amir Johnson\johnsam01    | 254  | C    | 30   | PHI  | 74   | 18   | 15.8 | 1.9  | 3.5  | 0.538 | 0.1  | 0.4  | 0.313 | 1.8  | 3.1  | 0.57  | 0.558 | 0.7  | 1.1  | 0.612 | 1.7  | 2.8  | 4.5  | 1.6  | 0.6  | 0.6  | 0.7  | 2.6  | 4.6  | 301  | PHI  | $2393887
Andre Drummond\drumman01  | 140  | C    | 24   | DET  | 78   | 78   | 33.7 | 6    | 11.3 | 0.529 | 0    | 0.1  | 0     | 6    | 11.2 | 0.536 | 0.529 | 3.1  | 5.1  | 0.605 | 5.1  | 10.9 | 16   | 3    | 1.5  | 1.6  | 2.6  | 3.2  | 15   | 25   | DET  | $25434263
Andre Iguodala\iguodan01  | 234  | SF   | 34   | GSW  | 64   | 7    | 25.3 | 2.3  | 5    | 0.463 | 0.5  | 1.8  | 0.282 | 1.8  | 3.2  | 0.567 | 0.514 | 0.9  | 1.4  | 0.632 | 0.8  | 3.1  | 3.8  | 3.3  | 0.8  | 0.6  | 1    | 1.5  | 6    | 71   | GSW  | $16000000
... (361 rows omitted)

In [180]:
clean_data['2018-19'] = clean_data.apply(convert_string_to_number, '2018-19')

In [181]:
clean_data

Player                    | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Aaron Gordon\gordoaa01    | 179  | PF   | 22   | ORL  | 58   | 57   | 32.9 | 6.5  | 14.9 | 0.434 | 2    | 5.9  | 0.336 | 4.5  | 9    | 0.497 | 0.5   | 2.7  | 3.9  | 0.698 | 1.5  | 6.4  | 7.9  | 2.3  | 1    | 0.8  | 1.8  | 1.9  | 17.6 | 40   | ORL  | 2.15909e+07
Al Horford\horfoal01      | 227  | C    | 31   | BOS  | 72   | 72   | 31.6 | 5.1  | 10.5 | 0.489 | 1.3  | 3.1  | 0.429 | 3.8  | 7.3  | 0.514 | 0.553 | 1.3  | 1.7  | 0.783 | 1.4  | 5.9  | 7.4  | 4.7  | 0.6  | 1.1  | 1.8  | 1.9  | 12.9 | 13   | BOS  | 2.89287e+07
Al Jefferson\jeffeal01    | 250  | C    | 33   | IND  | 36   | 1    | 13.4 | 3.1  | 5.8  | 0.534 | 0    | 0.1  | 0     | 3.1  | 5.7  | 0.541 | 0.534 | 0.8  | 1    | 0.833 | 0.8  | 3.1  | 4    | 0.8  | 0.4  | 0.6  | 0.6  | 1.8  | 7    | 235  | IND  | 4e+06
Al-Farouq Aminu\aminual01 | 11   | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3  | 179  | POR  | 6.95710e+06
Alec Burks\burksal01      | 78   | SG   | 26   | UTA  | 64   | 1    | 16.5 | 2.7  | 6.5  | 0.411 | 0.7  | 2.2  | 0.331 | 1.9  | 4.3  | 0.452 | 0.467 | 1.7  | 1.9  | 0.863 | 0.3  | 2.7  | 3    | 1    | 0.6  | 0.1  | 0.9  | 1.2  | 7.7  | 120  | UTA  | 1.15365e+07
Alex Len\lenal01          | 292  | C    | 24   | PHO  | 69   | 13   | 20.2 | 3.1  | 5.5  | 0.566 | 0    | 0    | 0.333 | 3.1  | 5.4  | 0.568 | 0.567 | 2.3  | 3.3  | 0.684 | 2.5  | 5    | 7.5  | 1.2  | 0.4  | 0.9  | 1.1  | 2.3  | 8.5  | 227  | ATL  | 4.35e+06
Allen Crabbe\crabbal01    | 114  | SG   | 25   | BRK  | 75   | 68   | 29.3 | 4.5  | 11   | 0.407 | 2.7  | 7.1  | 0.378 | 1.8  | 3.9  | 0.461 | 0.529 | 1.5  | 1.8  | 0.852 | 0.4  | 3.9  | 4.3  | 1.6  | 0.6  | 0.5  | 1    | 2.2  | 13.2 | 57   | BRK  | 1.85e+07
Amir Johnson\johnsam01    | 254  | C    | 30   | PHI  | 74   | 18   | 15.8 | 1.9  | 3.5  | 0.538 | 0.1  | 0.4  | 0.313 | 1.8  | 3.1  | 0.57  | 0.558 | 0.7  | 1.1  | 0.612 | 1.7  | 2.8  | 4.5  | 1.6  | 0.6  | 0.6  | 0.7  | 2.6  | 4.6  | 301  | PHI  | 2.39389e+06
Andre Drummond\drumman01  | 140  | C    | 24   | DET  | 78   | 78   | 33.7 | 6    | 11.3 | 0.529 | 0    | 0.1  | 0     | 6    | 11.2 | 0.536 | 0.529 | 3.1  | 5.1  | 0.605 | 5.1  | 10.9 | 16   | 3    | 1.5  | 1.6  | 2.6  | 3.2  | 15   | 25   | DET  | 2.54343e+07
Andre Iguodala\iguodan01  | 234  | SF   | 34   | GSW  | 64   | 7    | 25.3 | 2.3  | 5    | 0.463 | 0.5  | 1.8  | 0.282 | 1.8  | 3.2  | 0.567 | 0.514 | 0.9  | 1.4  | 0.632 | 0.8  | 3.1  | 3.8  | 3.3  | 0.8  | 0.6  | 1    | 1.5  | 6    | 71   | GSW  | 1.6e+07
... (361 rows omitted)

In [182]:
clean_data

Player                    | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Aaron Gordon\gordoaa01    | 179  | PF   | 22   | ORL  | 58   | 57   | 32.9 | 6.5  | 14.9 | 0.434 | 2    | 5.9  | 0.336 | 4.5  | 9    | 0.497 | 0.5   | 2.7  | 3.9  | 0.698 | 1.5  | 6.4  | 7.9  | 2.3  | 1    | 0.8  | 1.8  | 1.9  | 17.6 | 40   | ORL  | 2.15909e+07
Al Horford\horfoal01      | 227  | C    | 31   | BOS  | 72   | 72   | 31.6 | 5.1  | 10.5 | 0.489 | 1.3  | 3.1  | 0.429 | 3.8  | 7.3  | 0.514 | 0.553 | 1.3  | 1.7  | 0.783 | 1.4  | 5.9  | 7.4  | 4.7  | 0.6  | 1.1  | 1.8  | 1.9  | 12.9 | 13   | BOS  | 2.89287e+07
Al Jefferson\jeffeal01    | 250  | C    | 33   | IND  | 36   | 1    | 13.4 | 3.1  | 5.8  | 0.534 | 0    | 0.1  | 0     | 3.1  | 5.7  | 0.541 | 0.534 | 0.8  | 1    | 0.833 | 0.8  | 3.1  | 4    | 0.8  | 0.4  | 0.6  | 0.6  | 1.8  | 7    | 235  | IND  | 4e+06
Al-Farouq Aminu\aminual01 | 11   | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3  | 179  | POR  | 6.95710e+06
Alec Burks\burksal01      | 78   | SG   | 26   | UTA  | 64   | 1    | 16.5 | 2.7  | 6.5  | 0.411 | 0.7  | 2.2  | 0.331 | 1.9  | 4.3  | 0.452 | 0.467 | 1.7  | 1.9  | 0.863 | 0.3  | 2.7  | 3    | 1    | 0.6  | 0.1  | 0.9  | 1.2  | 7.7  | 120  | UTA  | 1.15365e+07
Alex Len\lenal01          | 292  | C    | 24   | PHO  | 69   | 13   | 20.2 | 3.1  | 5.5  | 0.566 | 0    | 0    | 0.333 | 3.1  | 5.4  | 0.568 | 0.567 | 2.3  | 3.3  | 0.684 | 2.5  | 5    | 7.5  | 1.2  | 0.4  | 0.9  | 1.1  | 2.3  | 8.5  | 227  | ATL  | 4.35e+06
Allen Crabbe\crabbal01    | 114  | SG   | 25   | BRK  | 75   | 68   | 29.3 | 4.5  | 11   | 0.407 | 2.7  | 7.1  | 0.378 | 1.8  | 3.9  | 0.461 | 0.529 | 1.5  | 1.8  | 0.852 | 0.4  | 3.9  | 4.3  | 1.6  | 0.6  | 0.5  | 1    | 2.2  | 13.2 | 57   | BRK  | 1.85e+07
Amir Johnson\johnsam01    | 254  | C    | 30   | PHI  | 74   | 18   | 15.8 | 1.9  | 3.5  | 0.538 | 0.1  | 0.4  | 0.313 | 1.8  | 3.1  | 0.57  | 0.558 | 0.7  | 1.1  | 0.612 | 1.7  | 2.8  | 4.5  | 1.6  | 0.6  | 0.6  | 0.7  | 2.6  | 4.6  | 301  | PHI  | 2.39389e+06
Andre Drummond\drumman01  | 140  | C    | 24   | DET  | 78   | 78   | 33.7 | 6    | 11.3 | 0.529 | 0    | 0.1  | 0     | 6    | 11.2 | 0.536 | 0.529 | 3.1  | 5.1  | 0.605 | 5.1  | 10.9 | 16   | 3    | 1.5  | 1.6  | 2.6  | 3.2  | 15   | 25   | DET  | 2.54343e+07
Andre Iguodala\iguodan01  | 234  | SF   | 34   | GSW  | 64   | 7    | 25.3 | 2.3  | 5    | 0.463 | 0.5  | 1.8  | 0.282 | 1.8  | 3.2  | 0.567 | 0.514 | 0.9  | 1.4  | 0.632 | 0.8  | 3.1  | 3.8  | 3.3  | 0.8  | 0.6  | 1    | 1.5  | 6    | 71   | GSW  | 1.6e+07
... (361 rows omitted)

In [183]:
players = clean_data.where('Player', are.not_equal_to('nan'))

In [184]:
players

Player                    | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Aaron Gordon\gordoaa01    | 179  | PF   | 22   | ORL  | 58   | 57   | 32.9 | 6.5  | 14.9 | 0.434 | 2    | 5.9  | 0.336 | 4.5  | 9    | 0.497 | 0.5   | 2.7  | 3.9  | 0.698 | 1.5  | 6.4  | 7.9  | 2.3  | 1    | 0.8  | 1.8  | 1.9  | 17.6 | 40   | ORL  | 2.15909e+07
Al Horford\horfoal01      | 227  | C    | 31   | BOS  | 72   | 72   | 31.6 | 5.1  | 10.5 | 0.489 | 1.3  | 3.1  | 0.429 | 3.8  | 7.3  | 0.514 | 0.553 | 1.3  | 1.7  | 0.783 | 1.4  | 5.9  | 7.4  | 4.7  | 0.6  | 1.1  | 1.8  | 1.9  | 12.9 | 13   | BOS  | 2.89287e+07
Al Jefferson\jeffeal01    | 250  | C    | 33   | IND  | 36   | 1    | 13.4 | 3.1  | 5.8  | 0.534 | 0    | 0.1  | 0     | 3.1  | 5.7  | 0.541 | 0.534 | 0.8  | 1    | 0.833 | 0.8  | 3.1  | 4    | 0.8  | 0.4  | 0.6  | 0.6  | 1.8  | 7    | 235  | IND  | 4e+06
Al-Farouq Aminu\aminual01 | 11   | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3  | 179  | POR  | 6.95710e+06
Alec Burks\burksal01      | 78   | SG   | 26   | UTA  | 64   | 1    | 16.5 | 2.7  | 6.5  | 0.411 | 0.7  | 2.2  | 0.331 | 1.9  | 4.3  | 0.452 | 0.467 | 1.7  | 1.9  | 0.863 | 0.3  | 2.7  | 3    | 1    | 0.6  | 0.1  | 0.9  | 1.2  | 7.7  | 120  | UTA  | 1.15365e+07
Alex Len\lenal01          | 292  | C    | 24   | PHO  | 69   | 13   | 20.2 | 3.1  | 5.5  | 0.566 | 0    | 0    | 0.333 | 3.1  | 5.4  | 0.568 | 0.567 | 2.3  | 3.3  | 0.684 | 2.5  | 5    | 7.5  | 1.2  | 0.4  | 0.9  | 1.1  | 2.3  | 8.5  | 227  | ATL  | 4.35e+06
Allen Crabbe\crabbal01    | 114  | SG   | 25   | BRK  | 75   | 68   | 29.3 | 4.5  | 11   | 0.407 | 2.7  | 7.1  | 0.378 | 1.8  | 3.9  | 0.461 | 0.529 | 1.5  | 1.8  | 0.852 | 0.4  | 3.9  | 4.3  | 1.6  | 0.6  | 0.5  | 1    | 2.2  | 13.2 | 57   | BRK  | 1.85e+07
Amir Johnson\johnsam01    | 254  | C    | 30   | PHI  | 74   | 18   | 15.8 | 1.9  | 3.5  | 0.538 | 0.1  | 0.4  | 0.313 | 1.8  | 3.1  | 0.57  | 0.558 | 0.7  | 1.1  | 0.612 | 1.7  | 2.8  | 4.5  | 1.6  | 0.6  | 0.6  | 0.7  | 2.6  | 4.6  | 301  | PHI  | 2.39389e+06
Andre Drummond\drumman01  | 140  | C    | 24   | DET  | 78   | 78   | 33.7 | 6    | 11.3 | 0.529 | 0    | 0.1  | 0     | 6    | 11.2 | 0.536 | 0.529 | 3.1  | 5.1  | 0.605 | 5.1  | 10.9 | 16   | 3    | 1.5  | 1.6  | 2.6  | 3.2  | 15   | 25   | DET  | 2.54343e+07
Andre Iguodala\iguodan01  | 234  | SF   | 34   | GSW  | 64   | 7    | 25.3 | 2.3  | 5    | 0.463 | 0.5  | 1.8  | 0.282 | 1.8  | 3.2  | 0.567 | 0.514 | 0.9  | 1.4  | 0.632 | 0.8  | 3.1  | 3.8  | 3.3  | 0.8  | 0.6  | 1    | 1.5  | 6    | 71   | GSW  | 1.6e+07
... (234 rows omitted)

***SETUP TO FIND LINEAR WEIGHTS:***

In [185]:
import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns
from datascience import Table, are
import numpy as np
from datascience import *

### Loading the Cleaned Data Set ###

In [186]:
players

Player                    | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Aaron Gordon\gordoaa01    | 179  | PF   | 22   | ORL  | 58   | 57   | 32.9 | 6.5  | 14.9 | 0.434 | 2    | 5.9  | 0.336 | 4.5  | 9    | 0.497 | 0.5   | 2.7  | 3.9  | 0.698 | 1.5  | 6.4  | 7.9  | 2.3  | 1    | 0.8  | 1.8  | 1.9  | 17.6 | 40   | ORL  | 2.15909e+07
Al Horford\horfoal01      | 227  | C    | 31   | BOS  | 72   | 72   | 31.6 | 5.1  | 10.5 | 0.489 | 1.3  | 3.1  | 0.429 | 3.8  | 7.3  | 0.514 | 0.553 | 1.3  | 1.7  | 0.783 | 1.4  | 5.9  | 7.4  | 4.7  | 0.6  | 1.1  | 1.8  | 1.9  | 12.9 | 13   | BOS  | 2.89287e+07
Al Jefferson\jeffeal01    | 250  | C    | 33   | IND  | 36   | 1    | 13.4 | 3.1  | 5.8  | 0.534 | 0    | 0.1  | 0     | 3.1  | 5.7  | 0.541 | 0.534 | 0.8  | 1    | 0.833 | 0.8  | 3.1  | 4    | 0.8  | 0.4  | 0.6  | 0.6  | 1.8  | 7    | 235  | IND  | 4e+06
Al-Farouq Aminu\aminual01 | 11   | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3  | 179  | POR  | 6.95710e+06
Alec Burks\burksal01      | 78   | SG   | 26   | UTA  | 64   | 1    | 16.5 | 2.7  | 6.5  | 0.411 | 0.7  | 2.2  | 0.331 | 1.9  | 4.3  | 0.452 | 0.467 | 1.7  | 1.9  | 0.863 | 0.3  | 2.7  | 3    | 1    | 0.6  | 0.1  | 0.9  | 1.2  | 7.7  | 120  | UTA  | 1.15365e+07
Alex Len\lenal01          | 292  | C    | 24   | PHO  | 69   | 13   | 20.2 | 3.1  | 5.5  | 0.566 | 0    | 0    | 0.333 | 3.1  | 5.4  | 0.568 | 0.567 | 2.3  | 3.3  | 0.684 | 2.5  | 5    | 7.5  | 1.2  | 0.4  | 0.9  | 1.1  | 2.3  | 8.5  | 227  | ATL  | 4.35e+06
Allen Crabbe\crabbal01    | 114  | SG   | 25   | BRK  | 75   | 68   | 29.3 | 4.5  | 11   | 0.407 | 2.7  | 7.1  | 0.378 | 1.8  | 3.9  | 0.461 | 0.529 | 1.5  | 1.8  | 0.852 | 0.4  | 3.9  | 4.3  | 1.6  | 0.6  | 0.5  | 1    | 2.2  | 13.2 | 57   | BRK  | 1.85e+07
Amir Johnson\johnsam01    | 254  | C    | 30   | PHI  | 74   | 18   | 15.8 | 1.9  | 3.5  | 0.538 | 0.1  | 0.4  | 0.313 | 1.8  | 3.1  | 0.57  | 0.558 | 0.7  | 1.1  | 0.612 | 1.7  | 2.8  | 4.5  | 1.6  | 0.6  | 0.6  | 0.7  | 2.6  | 4.6  | 301  | PHI  | 2.39389e+06
Andre Drummond\drumman01  | 140  | C    | 24   | DET  | 78   | 78   | 33.7 | 6    | 11.3 | 0.529 | 0    | 0.1  | 0     | 6    | 11.2 | 0.536 | 0.529 | 3.1  | 5.1  | 0.605 | 5.1  | 10.9 | 16   | 3    | 1.5  | 1.6  | 2.6  | 3.2  | 15   | 25   | DET  | 2.54343e+07
Andre Iguodala\iguodan01  | 234  | SF   | 34   | GSW  | 64   | 7    | 25.3 | 2.3  | 5    | 0.463 | 0.5  | 1.8  | 0.282 | 1.8  | 3.2  | 0.567 | 0.514 | 0.9  | 1.4  | 0.632 | 0.8  | 3.1  | 3.8  | 3.3  | 0.8  | 0.6  | 1    | 1.5  | 6    | 71   | GSW  | 1.6e+07
... (234 rows omitted)

**Separate table by Position:**

In [297]:
#PG
point_guard = players.where('Pos', are.equal_to('PG'))
point_guard

Player                    | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Chris Paul\paulch01       | 396  | PG   | 32   | HOU  | 58   | 58   | 31.8 | 6.3  | 13.8 | 0.46  | 2.5  | 6.5  | 0.38  | 3.8  | 7.2  | 0.532 | 0.55  | 3.5  | 3.8  | 0.919 | 0.6  | 4.8  | 5.4  | 7.9  | 1.7  | 0.2  | 2.2  | 2.4  | 18.6 | 2    | HOU  | 3.56542e+07
Cory Joseph\josepco01     | 269  | PG   | 26   | IND  | 82   | 17   | 27   | 3.1  | 7.4  | 0.424 | 0.8  | 2.3  | 0.353 | 2.3  | 5    | 0.457 | 0.479 | 0.9  | 1.1  | 0.745 | 0.5  | 2.7  | 3.2  | 3.2  | 1    | 0.2  | 1.1  | 1.7  | 7.9  | 161  | IND  | 7.945e+06
D.J. Augustin\augusdj01   | 24   | PG   | 30   | ORL  | 75   | 36   | 23.5 | 3.3  | 7.2  | 0.452 | 1.5  | 3.6  | 0.419 | 1.7  | 3.6  | 0.485 | 0.557 | 2.2  | 2.5  | 0.868 | 0.4  | 1.7  | 2.1  | 3.8  | 0.7  | 0    | 1.6  | 1.3  | 10.2 | 171  | ORL  | 7.25e+06
Damian Lillard\lillada01  | 298  | PG   | 27   | POR  | 73   | 73   | 36.6 | 8.5  | 19.4 | 0.439 | 3.1  | 8.6  | 0.361 | 5.4  | 10.8 | 0.501 | 0.519 | 6.8  | 7.4  | 0.916 | 0.8  | 3.6  | 4.5  | 6.6  | 1.1  | 0.4  | 2.8  | 1.6  | 26.9 | 14   | POR  | 2.79777e+07
Dante Exum\exumda01       | 153  | PG   | 22   | UTA  | 14   | 0    | 16.8 | 3    | 6.2  | 0.483 | 0.4  | 1.3  | 0.278 | 2.6  | 4.9  | 0.536 | 0.511 | 1.8  | 2.2  | 0.806 | 0.3  | 1.6  | 1.9  | 3.1  | 0.6  | 0.2  | 1.4  | 1.4  | 8.1  | 138  | UTA  | 9.6e+06
Darren Collison\collida01 | 104  | PG   | 30   | IND  | 69   | 64   | 29.2 | 4.6  | 9.2  | 0.495 | 1.4  | 3    | 0.468 | 3.2  | 6.3  | 0.508 | 0.571 | 1.8  | 2.1  | 0.882 | 0.5  | 2    | 2.6  | 5.3  | 1.3  | 0.2  | 1.2  | 1.8  | 12.4 | 135  | IND  | 1e+07
Dennis Schroder\schrode01 | 442  | PG   | 24   | ATL  | 67   | 67   | 31   | 7.5  | 17.1 | 0.436 | 1.1  | 3.9  | 0.29  | 6.3  | 13.2 | 0.48  | 0.469 | 3.4  | 4    | 0.849 | 0.7  | 2.4  | 3.1  | 6.2  | 1.1  | 0.1  | 2.7  | 2.2  | 19.4 | 74   | OKC  | 1.55e+07
Derrick Rose\rosede01     | 433  | PG   | 29   | TOT  | 25   | 7    | 16.8 | 3.2  | 7.4  | 0.435 | 0.3  | 1.2  | 0.233 | 3    | 6.2  | 0.474 | 0.454 | 1.6  | 1.8  | 0.87  | 0.5  | 0.9  | 1.4  | 1.5  | 0.3  | 0.2  | 1.4  | 0.6  | 8.4  | 324  | MIN  | 2.17626e+06
Devin Harris\harride01    | 197  | PG   | 34   | TOT  | 71   | 1    | 18.9 | 2.6  | 6.4  | 0.412 | 1.1  | 3.2  | 0.348 | 1.5  | 3.2  | 0.476 | 0.499 | 2    | 2.4  | 0.835 | 0.2  | 1.6  | 1.8  | 2.1  | 0.7  | 0.2  | 1.1  | 2.2  | 8.4  | 311  | DAL  | 2.39389e+06
Elfrid Payton\paytoel01   | 399  | PG   | 23   | TOT  | 63   | 63   | 28.7 | 5.2  | 10.6 | 0.493 | 0.5  | 1.5  | 0.326 | 4.7  | 9.1  | 0.52  | 0.516 | 1.7  | 2.7  | 0.649 | 1    | 3.3  | 4.3  | 6.2  | 1.3  | 0.3  | 2.7  | 2.3  | 12.7 | 270  | NOP  | 3e+06
... (34 rows omitted)

In [298]:
#SG
shooting_guard = players.where('Pos', are.equal_to('SG'))
shooting_guard

Player                   | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Alec Burks\burksal01     | 78   | SG   | 26   | UTA  | 64   | 1    | 16.5 | 2.7  | 6.5  | 0.411 | 0.7  | 2.2  | 0.331 | 1.9  | 4.3  | 0.452 | 0.467 | 1.7  | 1.9  | 0.863 | 0.3  | 2.7  | 3    | 1    | 0.6  | 0.1  | 0.9  | 1.2  | 7.7  | 120  | UTA  | 1.15365e+07
Allen Crabbe\crabbal01   | 114  | SG   | 25   | BRK  | 75   | 68   | 29.3 | 4.5  | 11   | 0.407 | 2.7  | 7.1  | 0.378 | 1.8  | 3.9  | 0.461 | 0.529 | 1.5  | 1.8  | 0.852 | 0.4  | 3.9  | 4.3  | 1.6  | 0.6  | 0.5  | 1    | 2.2  | 13.2 | 57   | BRK  | 1.85e+07
Andre Roberson\roberan03 | 429  | SG   | 26   | OKC  | 39   | 39   | 26.6 | 2.2  | 4.2  | 0.537 | 0.2  | 0.9  | 0.222 | 2    | 3.2  | 0.627 | 0.562 | 0.3  | 1    | 0.316 | 1.9  | 2.8  | 4.7  | 1.2  | 1.2  | 0.9  | 0.8  | 2.3  | 5    | 133  | OKC  | 1e+07
Austin Rivers\riverau01  | 428  | SG   | 25   | LAC  | 61   | 59   | 33.7 | 5.6  | 13.2 | 0.424 | 2.2  | 5.9  | 0.378 | 3.4  | 7.3  | 0.461 | 0.508 | 1.7  | 2.6  | 0.642 | 0.4  | 1.9  | 2.4  | 4    | 1.2  | 0.3  | 1.8  | 2.5  | 15.1 | 102  | WAS  | 1.265e+07
Avery Bradley\bradlav01  | 61   | SG   | 27   | TOT  | 46   | 46   | 31.2 | 5.6  | 13.6 | 0.414 | 1.6  | 4.4  | 0.369 | 4    | 9.2  | 0.436 | 0.474 | 1.4  | 1.8  | 0.768 | 0.5  | 2    | 2.5  | 2    | 1.1  | 0.2  | 2.2  | 2    | 14.3 | 112  | LAC  | 1.2e+07
Ben McLemore\mclembe01   | 331  | SG   | 24   | MEM  | 56   | 17   | 19.5 | 2.8  | 6.7  | 0.421 | 1    | 2.8  | 0.346 | 1.8  | 3.8  | 0.477 | 0.495 | 0.9  | 1.1  | 0.828 | 0.6  | 1.9  | 2.5  | 0.9  | 0.7  | 0.3  | 1.1  | 2.5  | 7.5  | 199  | SAC  | 5.46e+06
Bradley Beal\bealbr01    | 37   | SG   | 24   | WAS  | 82   | 82   | 36.3 | 8.3  | 18.1 | 0.46  | 2.4  | 6.5  | 0.375 | 5.9  | 11.6 | 0.507 | 0.527 | 3.6  | 4.5  | 0.791 | 0.7  | 3.7  | 4.4  | 4.5  | 1.2  | 0.4  | 2.6  | 2    | 22.6 | 23   | WAS  | 2.54343e+07
CJ McCollum\mccolcj01    | 322  | SG   | 26   | POR  | 81   | 81   | 36.1 | 8.2  | 18.6 | 0.443 | 2.3  | 5.9  | 0.397 | 5.9  | 12.7 | 0.465 | 0.506 | 2.6  | 3.1  | 0.836 | 0.7  | 3.3  | 4    | 3.4  | 1    | 0.4  | 1.9  | 2.1  | 21.4 | 19   | POR  | 2.57598e+07
Courtney Lee\leeco01     | 289  | SG   | 32   | NYK  | 76   | 69   | 30.4 | 4.5  | 9.9  | 0.454 | 1.5  | 3.7  | 0.406 | 3    | 6.3  | 0.482 | 0.529 | 1.5  | 1.6  | 0.919 | 0.4  | 2.6  | 2.9  | 2.4  | 1.1  | 0.2  | 1.1  | 1.4  | 12   | 107  | NYK  | 1.22538e+07
Danny Green\greenda02    | 186  | SG   | 30   | SAS  | 70   | 60   | 25.6 | 3.1  | 8    | 0.387 | 1.7  | 4.6  | 0.363 | 1.4  | 3.4  | 0.419 | 0.49  | 0.7  | 0.9  | 0.769 | 0.6  | 3    | 3.6  | 1.6  | 0.9  | 1.1  | 1.1  | 1.7  | 8.6  | 134  | TOR  | 1e+07
... (47 rows omitted)

In [299]:
#SF
small_forward = players.where('Pos', are.equal_to('SF'))
small_forward

Player                     | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Andre Iguodala\iguodan01   | 234  | SF   | 34   | GSW  | 64   | 7    | 25.3 | 2.3  | 5    | 0.463 | 0.5  | 1.8  | 0.282 | 1.8  | 3.2  | 0.567 | 0.514 | 0.9  | 1.4  | 0.632 | 0.8  | 3.1  | 3.8  | 3.3  | 0.8  | 0.6  | 1    | 1.5  | 6    | 71   | GSW  | 1.6e+07
Andrew Wiggins\wiggian01   | 514  | SF   | 22   | MIN  | 82   | 82   | 36.3 | 6.9  | 15.9 | 0.438 | 1.4  | 4.1  | 0.331 | 5.6  | 11.7 | 0.475 | 0.481 | 2.5  | 3.8  | 0.643 | 1    | 3.4  | 4.4  | 2    | 1.1  | 0.6  | 1.7  | 2    | 17.7 | 21   | MIN  | 2.54672e+07
Bojan Bogdanovic\bogdabo02 | 55   | SF   | 28   | IND  | 80   | 80   | 30.8 | 5.1  | 10.7 | 0.474 | 1.9  | 4.8  | 0.402 | 3.1  | 5.8  | 0.534 | 0.565 | 2.2  | 2.6  | 0.868 | 0.4  | 3    | 3.4  | 1.5  | 0.7  | 0.1  | 1.3  | 1.6  | 14.3 | 129  | IND  | 1.05e+07
C.J. Miles\milescj01       | 337  | SF   | 30   | TOR  | 70   | 3    | 19.1 | 3.2  | 8.6  | 0.379 | 2.3  | 6.5  | 0.361 | 0.9  | 2.1  | 0.434 | 0.516 | 1.2  | 1.4  | 0.835 | 0.3  | 1.9  | 2.2  | 0.8  | 0.5  | 0.3  | 0.6  | 1.9  | 10   | 155  | TOR  | 8.33333e+06
Chandler Parsons\parsoch01 | 392  | SF   | 29   | MEM  | 36   | 8    | 19.2 | 3    | 6.5  | 0.462 | 1.4  | 3.4  | 0.421 | 1.6  | 3.1  | 0.504 | 0.571 | 0.5  | 0.8  | 0.63  | 0.4  | 2.1  | 2.5  | 1.9  | 0.5  | 0.3  | 1    | 1.5  | 7.9  | 32   | MEM  | 2.41073e+07
Danilo Gallinari\gallida01 | 168  | SF   | 29   | LAC  | 21   | 21   | 32   | 4.9  | 12.2 | 0.398 | 1.7  | 5.3  | 0.324 | 3.1  | 6.9  | 0.455 | 0.469 | 3.9  | 4.1  | 0.931 | 0.4  | 4.4  | 4.8  | 2    | 0.6  | 0.5  | 1.2  | 1.4  | 15.3 | 41   | LAC  | 2.15876e+07
Dante Cunningham\cunnida01 | 119  | SF   | 30   | TOT  | 73   | 25   | 21.4 | 2.2  | 5    | 0.451 | 0.8  | 2.3  | 0.345 | 1.5  | 2.7  | 0.541 | 0.53  | 0.4  | 0.7  | 0.596 | 1.1  | 3    | 4.1  | 0.7  | 0.5  | 0.4  | 0.5  | 1.5  | 5.7  | 292  | SAS  | 2.5e+06
Darius Miller\milleda01    | 338  | SF   | 27   | NOP  | 82   | 3    | 23.7 | 2.6  | 5.9  | 0.444 | 1.8  | 4.4  | 0.411 | 0.8  | 1.6  | 0.535 | 0.594 | 0.7  | 0.8  | 0.866 | 0.2  | 1.8  | 2    | 1.4  | 0.3  | 0.2  | 0.7  | 2    | 7.8  | 323  | NOP  | 2.205e+06
DeMarre Carroll\carrode01  | 86   | SF   | 31   | BRK  | 73   | 73   | 29.9 | 4.5  | 10.8 | 0.414 | 2    | 5.4  | 0.371 | 2.5  | 5.4  | 0.457 | 0.506 | 2.6  | 3.4  | 0.764 | 1.4  | 5.1  | 6.6  | 2    | 0.8  | 0.4  | 1.4  | 2.3  | 13.5 | 75   | BRK  | 1.54e+07
Doug McDermott\mcderdo01   | 326  | SF   | 26   | TOT  | 81   | 4    | 21.8 | 2.9  | 6.3  | 0.467 | 1.1  | 2.7  | 0.426 | 1.8  | 3.6  | 0.497 | 0.557 | 0.8  | 1    | 0.79  | 0.5  | 2    | 2.5  | 1    | 0.2  | 0.2  | 0.7  | 1.5  | 7.8  | 168  | IND  | 7.33333e+06
... (29 rows omitted)

In [300]:
#PF
power_forward = players.where('Pos', are.equal_to('PF'))
power_forward

Player                     | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Aaron Gordon\gordoaa01     | 179  | PF   | 22   | ORL  | 58   | 57   | 32.9 | 6.5  | 14.9 | 0.434 | 2    | 5.9  | 0.336 | 4.5  | 9    | 0.497 | 0.5   | 2.7  | 3.9  | 0.698 | 1.5  | 6.4  | 7.9  | 2.3  | 1    | 0.8  | 1.8  | 1.9  | 17.6 | 40   | ORL  | 2.15909e+07
Al-Farouq Aminu\aminual01  | 11   | PF   | 27   | POR  | 69   | 67   | 30   | 3.3  | 8.4  | 0.395 | 1.8  | 4.9  | 0.369 | 1.5  | 3.5  | 0.432 | 0.503 | 0.9  | 1.2  | 0.738 | 1.4  | 6.2  | 7.6  | 1.2  | 1.1  | 0.6  | 1.1  | 2    | 9.3  | 179  | POR  | 6.95710e+06
Anthony Davis\davisan02    | 122  | PF   | 24   | NOP  | 75   | 75   | 36.4 | 10.4 | 19.5 | 0.534 | 0.7  | 2.2  | 0.34  | 9.7  | 17.3 | 0.558 | 0.552 | 6.6  | 8    | 0.828 | 2.5  | 8.6  | 11.1 | 2.3  | 1.5  | 2.6  | 2.2  | 2.1  | 28.1 | 24   | NOP  | 2.54343e+07
Anthony Tolliver\tollian01 | 482  | PF   | 32   | DET  | 79   | 14   | 22.2 | 2.8  | 6    | 0.464 | 2    | 4.6  | 0.436 | 0.8  | 1.4  | 0.561 | 0.632 | 1.3  | 1.7  | 0.797 | 0.6  | 2.4  | 3.1  | 1.1  | 0.4  | 0.3  | 0.7  | 1.8  | 8.9  | 195  | MIN  | 5.75e+06
Blake Griffin\griffbl01    | 191  | PF   | 28   | TOT  | 58   | 58   | 34   | 7.5  | 17.2 | 0.438 | 1.9  | 5.6  | 0.345 | 5.6  | 11.6 | 0.482 | 0.493 | 4.5  | 5.7  | 0.785 | 1.3  | 6.1  | 7.4  | 5.8  | 0.7  | 0.3  | 2.8  | 2.4  | 21.4 | 5    | DET  | 3.20889e+07
Carmelo Anthony\anthoca01  | 17   | PF   | 33   | OKC  | 78   | 78   | 32.1 | 6.1  | 15   | 0.404 | 2.2  | 6.1  | 0.357 | 3.9  | 8.9  | 0.437 | 0.476 | 1.9  | 2.5  | 0.767 | 0.9  | 4.9  | 5.8  | 1.3  | 0.6  | 0.6  | 1.3  | 2.5  | 16.2 | 20   | ATL  | 2.55343e+07
Carmelo Anthony\anthoca01  | 17   | PF   | 33   | OKC  | 78   | 78   | 32.1 | 6.1  | 15   | 0.404 | 2.2  | 6.1  | 0.357 | 3.9  | 8.9  | 0.437 | 0.476 | 1.9  | 2.5  | 0.767 | 0.9  | 4.9  | 5.8  | 1.3  | 0.6  | 0.6  | 1.3  | 2.5  | 16.2 | 308  | HOU  | 2.39389e+06
Darrell Arthur\arthuda01   | 21   | PF   | 29   | DEN  | 19   | 1    | 7.4  | 1.2  | 2.5  | 0.468 | 0.4  | 1.2  | 0.348 | 0.7  | 1.3  | 0.583 | 0.553 | 0.1  | 0.2  | 0.667 | 0.2  | 0.6  | 0.8  | 0.5  | 0.4  | 0.2  | 0.8  | 1.2  | 2.8  | 166  | PHO  | 7.46491e+06
Draymond Green\greendr01   | 187  | PF   | 27   | GSW  | 70   | 70   | 32.7 | 4    | 8.8  | 0.454 | 1.1  | 3.7  | 0.301 | 2.9  | 5.2  | 0.562 | 0.516 | 1.9  | 2.5  | 0.775 | 1.1  | 6.6  | 7.6  | 7.3  | 1.4  | 1.3  | 2.9  | 2.6  | 11   | 61   | GSW  | 1.74696e+07
Ersan Ilyasova\ilyaser01   | 235  | PF   | 30   | TOT  | 69   | 43   | 25.1 | 4    | 8.9  | 0.452 | 1.3  | 3.7  | 0.36  | 2.7  | 5.2  | 0.517 | 0.526 | 1.5  | 2    | 0.778 | 1.7  | 4.2  | 5.9  | 1.3  | 0.9  | 0.4  | 1    | 2.3  | 10.9 | 177  | MIL  | 7e+06
... (35 rows omitted)

In [301]:
#C
center = players.where('Pos', are.equal_to('C'))
center

Player                    | Rk   | Pos  | Age  | Tm   | G    | GS   | MP   | FG   | FGA  | FG%   | 3P   | 3PA  | 3P%   | 2P   | 2PA  | 2P%   | eFG%  | FT   | FTA  | FT%   | ORB  | DRB  | TRB  | AST  | STL  | BLK  | TOV  | PF   | PS/G | Rk_2 | Tm_2 | 2018-19
Al Horford\horfoal01      | 227  | C    | 31   | BOS  | 72   | 72   | 31.6 | 5.1  | 10.5 | 0.489 | 1.3  | 3.1  | 0.429 | 3.8  | 7.3  | 0.514 | 0.553 | 1.3  | 1.7  | 0.783 | 1.4  | 5.9  | 7.4  | 4.7  | 0.6  | 1.1  | 1.8  | 1.9  | 12.9 | 13   | BOS  | 2.89287e+07
Al Jefferson\jeffeal01    | 250  | C    | 33   | IND  | 36   | 1    | 13.4 | 3.1  | 5.8  | 0.534 | 0    | 0.1  | 0     | 3.1  | 5.7  | 0.541 | 0.534 | 0.8  | 1    | 0.833 | 0.8  | 3.1  | 4    | 0.8  | 0.4  | 0.6  | 0.6  | 1.8  | 7    | 235  | IND  | 4e+06
Alex Len\lenal01          | 292  | C    | 24   | PHO  | 69   | 13   | 20.2 | 3.1  | 5.5  | 0.566 | 0    | 0    | 0.333 | 3.1  | 5.4  | 0.568 | 0.567 | 2.3  | 3.3  | 0.684 | 2.5  | 5    | 7.5  | 1.2  | 0.4  | 0.9  | 1.1  | 2.3  | 8.5  | 227  | ATL  | 4.35e+06
Amir Johnson\johnsam01    | 254  | C    | 30   | PHI  | 74   | 18   | 15.8 | 1.9  | 3.5  | 0.538 | 0.1  | 0.4  | 0.313 | 1.8  | 3.1  | 0.57  | 0.558 | 0.7  | 1.1  | 0.612 | 1.7  | 2.8  | 4.5  | 1.6  | 0.6  | 0.6  | 0.7  | 2.6  | 4.6  | 301  | PHI  | 2.39389e+06
Andre Drummond\drumman01  | 140  | C    | 24   | DET  | 78   | 78   | 33.7 | 6    | 11.3 | 0.529 | 0    | 0.1  | 0     | 6    | 11.2 | 0.536 | 0.529 | 3.1  | 5.1  | 0.605 | 5.1  | 10.9 | 16   | 3    | 1.5  | 1.6  | 2.6  | 3.2  | 15   | 25   | DET  | 2.54343e+07
Aron Baynes\baynear01     | 35   | C    | 31   | BOS  | 81   | 67   | 18.3 | 2.6  | 5.5  | 0.471 | 0    | 0.3  | 0.143 | 2.6  | 5.2  | 0.487 | 0.474 | 0.7  | 1    | 0.756 | 1.6  | 3.8  | 5.4  | 1.1  | 0.3  | 0.6  | 1    | 2.5  | 6    | 210  | BOS  | 5.1936e+06
Bismack Biyombo\biyombi01 | 48   | C    | 25   | ORL  | 82   | 25   | 18.2 | 2.2  | 4.3  | 0.52  | 0    | 0    | 0     | 2.2  | 4.3  | 0.521 | 0.52  | 1.2  | 1.9  | 0.65  | 1.8  | 4    | 5.7  | 0.8  | 0.3  | 1.2  | 1    | 1.9  | 5.7  | 66   | CHO  | 1.7e+07
Brook Lopez\lopezbr01     | 302  | C    | 29   | LAL  | 74   | 72   | 23.4 | 5    | 10.7 | 0.465 | 1.5  | 4.4  | 0.345 | 3.5  | 6.3  | 0.549 | 0.536 | 1.5  | 2.1  | 0.703 | 1    | 3    | 4    | 1.7  | 0.4  | 1.3  | 1.3  | 2.6  | 13   | 253  | MIL  | 3.382e+06
Channing Frye\fryech01    | 166  | C    | 34   | TOT  | 53   | 1    | 13.1 | 2    | 4    | 0.491 | 0.7  | 2.2  | 0.339 | 1.2  | 1.8  | 0.67  | 0.583 | 0.3  | 0.4  | 0.895 | 0.2  | 2.3  | 2.5  | 0.7  | 0.3  | 0.2  | 0.4  | 1.3  | 5    | 312  | CLE  | 2.39389e+06
Clint Capela\capelca01    | 85   | C    | 23   | HOU  | 74   | 74   | 27.5 | 6    | 9.1  | 0.652 | 0    | 0    | 0     | 6    | 9.1  | 0.653 | 0.652 | 1.9  | 3.5  | 0.56  | 3.3  | 7.6  | 10.8 | 0.9  | 0.8  | 1.9  | 1.4  | 2.5  | 13.9 | 76   | HOU  | 1.52931e+07
... (48 rows omitted)

**Tables of statistic sum per position, including salary average for that position:**

In [302]:
# input table of players for each position, outputs table with sums of each category of statistic:
def stat_sums_table(table):
    fg_sum = np.sum(table.column('FG'))
    fga_sum = np.sum(table.column('FGA'))
    fgpct_sum = np.sum(table.column('FG%'))
    three_p_sum = np.sum(table.column('3P'))
    three_pa_sum = np.sum(table.column('3PA'))
    three_ppct_sum = np.sum(table.column('3P%'))
    two_p_sum = np.sum(table.column('2P'))
    two_pa_sum = np.sum(table.column('2PA'))
    two_ppct_sum = np.sum(table.column('2P%'))
    efgpct_sum = np.sum(table.column('eFG%'))
    orb_sum = np.sum(table.column('ORB'))
    drb_sum = np.sum(table.column('DRB'))
    trb_sum = np.sum(table.column('TRB'))
    ast_sum = np.sum(table.column('AST'))
    stl_sum = np.sum(table.column('STL'))
    blk_sum = np.sum(table.column('BLK'))
    tov_sum = np.sum(table.column('TOV'))
    pf_sum = np.sum(table.column('PF'))
    pts_sum = np.sum(table.column('PS/G'))
    salary_sum = np.sum(table.column('2018-19'))

    stat_cats = ('FG', 'FGA', 'FG%', '3P', '3PA', '3P%', '2P', '2PA', '2P%', 'eFG%', 'ORB', 'DRB', 
                    'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PPG', '2018-19 Salary')

    stat_sums = (fg_sum, fga_sum, fgpct_sum, three_p_sum, three_pa_sum, three_ppct_sum,
                  two_p_sum, two_pa_sum, two_ppct_sum, efgpct_sum, orb_sum, drb_sum,
                  trb_sum, ast_sum, stl_sum, blk_sum, tov_sum, pf_sum, pts_sum, salary_sum)
    return Table().with_columns('Category', stat_cats, 'Sums', stat_sums)

In [386]:
point_guard_sums = stat_sums_table(point_guard)
point_guard_sums.show()

In [308]:
shooting_guard_sums = stat_sums_table(shooting_guard)
power_forward_sums = stat_sums_table(power_forward)
small_forward_sums = stat_sums_table(small_forward)
center_sums = stat_sums_table(center)

In [309]:
shooting_guard_sums = shooting_guard_sums.sort('Sums', descending=True)
power_forward_sums = power_forward_sums.sort('Sums', descending=True)
small_forward_sums = small_forward_sums.sort('Sums', descending=True)
center_sums = center_sums.sort('Sums', descending=True)

**Tables of statistic averages per position, including salary average for that position:**

In [305]:
# input table of players for each position, outputs table with sums of each category of statistic:
def stat_means_table(table):
    fg = np.mean(table.column('FG'))
    fga = np.mean(table.column('FGA'))
    fgpct = np.mean(table.column('FG%'))
    three_p = np.mean(table.column('3P'))
    three_pa = np.mean(table.column('3PA'))
    three_ppct = np.mean(table.column('3P%'))
    two_p = np.mean(table.column('2P'))
    two_pa = np.mean(table.column('2PA'))
    two_ppct = np.mean(table.column('2P%'))
    efgpct = np.mean(table.column('eFG%'))
    orb = np.mean(table.column('ORB'))
    drb = np.mean(table.column('DRB'))
    trb = np.mean(table.column('TRB'))
    ast = np.mean(table.column('AST'))
    stl = np.mean(table.column('STL'))
    blk = np.mean(table.column('BLK'))
    tov = np.mean(table.column('TOV'))
    pf = np.mean(table.column('PF'))
    pts = np.mean(table.column('PS/G'))
    salary = np.mean(table.column('2018-19'))

    stat_cats = ('FG', 'FGA', 'FG%', '3P', '3PA', '3P%', '2P', '2PA', '2P%', 'eFG%', 'ORB', 'DRB', 
                    'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PPG', '2018-19 Salary')

    stat_averages = (fg, fga, fgpct, three_p, three_pa, three_ppct,
                  two_p, two_pa, two_ppct, efgpct, orb, drb,
                  trb, ast, stl, blk, tov, pf, pts, salary)
    return Table().with_columns('Category', stat_cats, 'Means', stat_averages)

In [387]:
point_guard_means = stat_means_table(point_guard)
point_guard_means.show()

In [307]:
shooting_guard_means = stat_means_table(shooting_guard)
power_forward_means = stat_means_table(power_forward)
small_forward_means = stat_means_table(small_forward)
center_means = stat_means_table(center)

In [310]:
shooting_guard_means = shooting_guard_means.sort('Means', descending=True)
power_forward_means = power_forward_means.sort('Means', descending=True)
small_forward_means = small_forward_means.sort('Means', descending=True)
center_means = center_means.sort('Means', descending=True)

### Analyze the rank of the positions' means per category. 

In [315]:
# change the 'means' column title to 'position means'
def change_title(table, position):
    table = table.relabel('Means', position + ' Means')

In [338]:
# join means columns from all tables into one to compare:
all_position_means = point_guard_means.join('Category', shooting_guard_means).join('Category', power_forward_means).join('Category', small_forward_means).join('Category', center_means)
all_position_means

Category       | PG Means    | SG Means    | PF Means    | SF Means   | C Means
2018-19 Salary | 1.20346e+07 | 1.11578e+07 | 1.30362e+07 | 1.2044e+07 | 1.15952e+07
2P             | 3.21591     | 2.94386     | 3.32444     | 2.35641    | 3.33793
2P%            | 0.482318    | 0.482667    | 0.507067    | 0.508821   | 0.5475
2PA            | 6.64545     | 6.04211     | 6.35778     | 4.63333    | 6.00862
3P             | 1.33636     | 1.74737     | 1.15778     | 1.18205    | 0.306897
3P%            | 0.343386    | 0.364526    | nan         | nan        | nan
3PA            | 3.64545     | 4.71404     | 3.15333     | 3.25128    | 0.901724
AST            | 4.61818     | 2.65263     | 1.95111     | 1.66923    | 1.43103
BLK            | 0.218182    | 0.335088    | 0.577778    | 0.361538   | 0.827586
DRB            | 2.53864     | 2.67193     | 4.24889     | 2.80769    | 4.67414
... (10 rows omitted)

In [341]:
all_position_means.show()

### After analying which statistic categories are most influential per position, we find the linear weights###

**Statistics chosen per position. Small Forward position contains more categories as they are versatile and aren't defined by any one statistic**

Point Guard: AST, PPG, TOV, 3P%, STL 
Shooting Guard: AST, PPG, TOV, 3P%, STL 
Small Forward: 2P%, 3P%, BLK, DRB, FG%, ORD, TRB
Power Forward: DRB, BLK, FG%, ORD, TRB
Center: 2P%, BLK, FG%, ORB, TRB

### Use the sum tables per position to calculate the weight as such: ###
For some ***x*** position and some ***n*** statistic category:

Weight of ***n*** for position ***x*** = sum(total salary for all ***x***) / sum(total ***n*** for all ***x***)



For example, to find the weight of points for centers: 

Weight of points for centers = total salary of all centers / total points scored by all centers

In [388]:
point_guard_means.show()

In [397]:
# create a table with the sums of only the chosen statistics
pg_chosen = point_guard_sums.take((13, 18, 16, 5, 14, 19))
sg_chosen = shooting_guard_sums.take((1, 9, 12, 18, 13, 0))
sf_chosen = small_forward_sums.take((17, 19, 8, 11, 18, 15, 6, 1))
pf_chosen = power_forward_sums.take((7, 16, 19, 14, 5, 1))
c_chosen = center_sums.take((15, 14, 18, 10, 4, 1))
#Point Guard: AST, PPG, TOV, 3P%, STL 
pg_chosen

Category       | Sums
AST            | 203.2
PPG            | 562.7
TOV            | 82.1
3P%            | 15.109
STL            | 40.6
2018-19 Salary | 5.29523e+08

In [398]:
def calculate_weight(sums_table, stat_cat_title):
    salary = sums_table.where('Category', are.equal_to('2018-19 Salary')).column('Sums').item(0)
    category_value = sums_table.where('Category', are.equal_to(stat_cat_title)).column('Sums').item(0)
    return salary/category_value

In [399]:
def table_with_weights(sums_table):
    salary = sums_table.where('Category', are.equal_to('2018-19 Salary')).column('Sums').item(0)
    weight_array = []
    for i in np.arange(sums_table.num_rows):
        category_value = sums_table.column('Sums').item(i)
        weight_array.append(salary/category_value)
    return sums_table.with_column('Weights', weight_array)

In [400]:
pg_weights = table_with_weights(pg_chosen)
pg_weights

Category       | Sums        | Weights
AST            | 203.2       | 2.60592e+06
PPG            | 562.7       | 941040
TOV            | 82.1        | 6.44974e+06
3P%            | 15.109      | 3.50469e+07
STL            | 40.6        | 1.30424e+07
2018-19 Salary | 5.29523e+08 | 1

In [401]:
sg_weights = table_with_weights(sg_chosen)
pf_weights = table_with_weights(pf_chosen)
sf_weights = table_with_weights(sf_chosen)
c_weights = table_with_weights(c_chosen)

### Analyze and Reflect on weights and decide which are important:###

In [402]:
pg_weights.show()

The formula for point guard salary is given by,

Salary = 2.6 * AST + 0.94 * PPG - 6.4 * TOV + 35 * 3P% + 13 * STL

In [380]:
sg_weights.show()

The formula for shooting guard salary is given by,

Salary = 4.2 * AST + 0.85 * PPG - 7.2 * TOV + 30 * 3P% + 11 * STL

In [381]:
sf_weights.show()

The formula for small forward salary is given by,

Salary = 7.2 * AST + 2.3 * 2P% + 33 * BLK + 4.2 * DRB + 27 * FG% + 18.9 * ORB - 3.5 * TRB

In [382]:
pf_weights.show()

The formula for power forward salary is given by,

Salary = 3 * DRB + 22.5 * BLK + 28.3 * FG% + 11.2 * ORB - 2.4 * TRB

In [383]:
c_weights.show()

The formula for center salary is given by,

Salary = 21.1 * 2P% + 14 * BLK - 22.2 * FG% + 6.1 * ORB - 1.7 * TRB

### WEIGHTED SALARY FORMULAS FOR EACH POSITION:###

***Point Gaurd Salary:***
<br>
***Salary*** = 1.3 * ***AST*** + 0.77 * ***PPG*** - 3.7 * ***TOV*** + 2.9 * ***3P%*** + 6.8 * ***STL*** - 6.42

***Shooting Guard Salary:***
<br>
***Salary*** = 0.03 * ***3P*** + 1.86 * ***STL*** - 3.2 * ***AST*** + -0.99 * ***TOV*** + 7 * ***eFG%*** - 1.2

***Small Forward Salary:***
<br>
***Salary*** = 32.6 * ***2P%*** + 2.7 * ***BLK*** + 1.5 * ***TRB*** + 2.4 * ***3P*** - 1.3 * ***PF*** - 10.8

***Power Forward Salary:***
<br>
***Salary*** = -3.9 * ***TOV*** + 2.4 * ***BLK*** + 3 * ***AST*** + 1.8 * ***2P*** - 0.2 * ***TRB*** + 4.3

***Center Salary:***
<br>
***Salary*** = 0.4 * ***PPG*** + 0.4 * ***BLK*** + 0.7 * ***DRB*** + 6.9* ***FG&*** + 0.72

## Regression Analysis:###
    
### Intution and Execution of Analysis: ###
    1. The goal of the contract value equation was to single out factors that were important features of a certain 
       position's play, so that those factors can determine how much a player would earn. 
    2. To locate such factors, the sums of the statistics from the player tables were compared to each of the 
       other positions. Some statistical means were simply greater among certain position, and that indicated the 
       important of a factor to a position.  
    3. After find 5 of the most influential statistics, the linear regression was executed. The statistical means 
       were weighed against the average salary of all players in a certain position. 
    4. Some baseline was to be determined for all positions, which would function as an intercept in our 
       linear regression. 
    5. Negative statistic categories (such as turnovers for point guards) would be subtracted from positive 
       statistics (for example, points for shooting guards) and would be nullified with the intercept to result in        a fair salary. 
       
### Conclusions: ###
    1. During our calculations, we reasoned that the Small Forward was the most versatile player. In other words,        he wasn't strongly categorized and determined by any statistic. The Small Forward would be able to do a lot        of thing on the court, and this made sense- Lebron James is a small forward. 
    2. The value of the Point Guard and Shooting Guard are determined by the same statistics. 
    3. It is better to evalue a player's value by the 3-pointers or 2-pointers (essentially, the field-goals) he          makes because percents can negatively influence a player's value when it actually should add value. 
    4. Our model is not perfect; its accuracy compares to how fairly players are paid in the NBA.  
    